hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -372,7 +372,7 @@ protected static long averageBytesPerRecord(HoodieTimeline commitTimeline, Hoodi
     long avgSize = hoodieWriteConfig.getCopyOnWriteRecordSizeEstimate();
     long fileSizeThreshold = (long) (hoodieWriteConfig.getRecordSizeEstimationThreshold() * hoodieWriteConfig.getParquetSmallFileLimit());
     try {
-      if (!commitTimeline.empty()) {
+      if (hoodieWriteConfig.getRecordSizeEstimationThreshold() > 0 && !commitTimeline.empty()) {
         // Go over the reverse ordered commits to get a more recent estimate of average record size.
         Iterator<HoodieInstant> instants = commitTimeline.getReverseOrderedInstants().iterator();
```

Comment:
The default value already greater than 0

Reply:
> data size is fixed in some sense, more the hudi community did not encourage adding a more boolean variable to control whether to use the last commit avgSize, then we use the estimation threshold, when it is less than 0, we use the default estimate record size.

Did you mean the records size Variance is large ？

Reply:
hi, we found that the result of totalBytesWritten/totalRecordsWritten is small when the last commit, but the next commit record number is very large, then the data files will become very large, in our case the data file will come to 600~700M.

More, our record size is fixed about 100M, so we don't want to use the above algorithm to help calculate avgsize and use the default value of the estimated record size which is set by users.

Although the default value of the estimation threshold is 1.0, in fact, I want to extend the usage of this property and then don't need to add another boolean property to control this, when we set it to less than 0, then could use the default size not using the totalBytesWritten/totalRecordsWritten.

Reply:
> hi, we found that the result of totalBytesWritten/totalRecordsWritten is small when the last commit, but the next commit record number is very large

Got you, can you explain why the record size estimation from the first commit so small here ?

Reply:
>  if (hoodieWriteConfig.getRecordSizeEstimationThreshold() > 0 && !commitTimeline.empty()) 

Shouldn't we first determine if the default value is adjusted (org.apache.hudi.config.HoodieCompactionConfig#_COPY_ON_WRITE_RECORD_SIZE_ESTIMATE)?  I think this is the first priority. 

 Imagine, according to the logic adjusted above, that is, you can only set avgSize to a fixed 1024 (the default size) by adjusting the threshold, but not the ability to let users customize avgSize according to their personalized tasks. 

If I have misunderstood, please let me know ,thanks

@danny0405 @honeyaya @codope @nsivabalan